<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />

<meta name="viewport" content="width=device-width, initial-scale=1" />

<meta name="author" content="Shariq Mohammed" />

<meta name="date" content="2020-12-08" />

<title>RADIOHEAD Example</title>

<script>// Hide empty <a> tag within highlighted CodeBlock for screen reader accessibility (see https://github.com/jgm/pandoc/issues/6352#issuecomment-626106786) -->
// v0.0.1
// Written by JooYoung Seo (jooyoung@psu.edu) and Atsushi Yasumoto on June 1st, 2020.

document.addEventListener('DOMContentLoaded', function() {
  const codeList = document.getElementsByClassName("sourceCode");
  for (var i = 0; i < codeList.length; i++) {
    var linkList = codeList[i].getElementsByTagName('a');
    for (var j = 0; j < linkList.length; j++) {
      if (linkList[j].innerHTML === "") {
        linkList[j].setAttribute('aria-hidden', 'true');
      }
    }
  }
});
</script>
<style type="text/css">
a.anchor-section {margin-left: 10px; visibility: hidden; color: inherit;}
a.anchor-section::before {content: '#';}
.hasAnchor:hover a.anchor-section {visibility: visible;}
</style>
<script>// Anchor sections v1.0 written by Atsushi Yasumoto on Oct 3rd, 2020.
document.addEventListener('DOMContentLoaded', function() {
  // Do nothing if AnchorJS is used
  if (typeof window.anchors === 'object' && anchors.hasOwnProperty('hasAnchorJSLink')) {
    return;
  }

  const h = document.querySelectorAll('h1, h2, h3, h4, h5, h6');

  // Do nothing if sections are already anchored
  if (Array.from(h).some(x => x.classList.contains('hasAnchor'))) {
    return null;
  }

  // Use section id when pandoc runs with --section-divs
  const section_id = function(x) {
    return ((x.classList.contains('section') || (x.tagName === 'SECTION'))
            ? x.id : '');
  };

  // Add anchors
  h.forEach(function(x) {
    const id = x.id || section_id(x.parentElement);
    if (id === '') {
      return null;
    }
    let anchor = document.createElement('a');
    anchor.href = '#' + id;
    anchor.classList = ['anchor-section'];
    x.classList.add('hasAnchor');
    x.appendChild(anchor);
  });
});
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>


<style type="text/css">code{white-space: pre;}</style>
<style type="text/css" data-origin="pandoc">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    for (var j = 0; j < rules.length; j++) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") continue;
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') continue;
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>



<style type="text/css">body {
background-color: #fff;
margin: 1em auto;
max-width: 700px;
overflow: visible;
padding-left: 2em;
padding-right: 2em;
font-family: "Open Sans", "Helvetica Neue", Helvetica, Arial, sans-serif;
font-size: 14px;
line-height: 1.35;
}
#TOC {
clear: both;
margin: 0 0 10px 10px;
padding: 4px;
width: 400px;
border: 1px solid #CCCCCC;
border-radius: 5px;
background-color: #f6f6f6;
font-size: 13px;
line-height: 1.3;
}
#TOC .toctitle {
font-weight: bold;
font-size: 15px;
margin-left: 5px;
}
#TOC ul {
padding-left: 40px;
margin-left: -1.5em;
margin-top: 5px;
margin-bottom: 5px;
}
#TOC ul ul {
margin-left: -2em;
}
#TOC li {
line-height: 16px;
}
table {
margin: 1em auto;
border-width: 1px;
border-color: #DDDDDD;
border-style: outset;
border-collapse: collapse;
}
table th {
border-width: 2px;
padding: 5px;
border-style: inset;
}
table td {
border-width: 1px;
border-style: inset;
line-height: 18px;
padding: 5px 5px;
}
table, table th, table td {
border-left-style: none;
border-right-style: none;
}
table thead, table tr.even {
background-color: #f7f7f7;
}
p {
margin: 0.5em 0;
}
blockquote {
background-color: #f6f6f6;
padding: 0.25em 0.75em;
}
hr {
border-style: solid;
border: none;
border-top: 1px solid #777;
margin: 28px 0;
}
dl {
margin-left: 0;
}
dl dd {
margin-bottom: 13px;
margin-left: 13px;
}
dl dt {
font-weight: bold;
}
ul {
margin-top: 0;
}
ul li {
list-style: circle outside;
}
ul ul {
margin-bottom: 0;
}
pre, code {
background-color: #f7f7f7;
border-radius: 3px;
color: #333;
white-space: pre-wrap; 
}
pre {
border-radius: 3px;
margin: 5px 0px 10px 0px;
padding: 10px;
}
pre:not([class]) {
background-color: #f7f7f7;
}
code {
font-family: Consolas, Monaco, 'Courier New', monospace;
font-size: 85%;
}
p > code, li > code {
padding: 2px 0px;
}
div.figure {
text-align: center;
}
img {
background-color: #FFFFFF;
padding: 2px;
border: 1px solid #DDDDDD;
border-radius: 3px;
border: 1px solid #CCCCCC;
margin: 0 5px;
}
h1 {
margin-top: 0;
font-size: 35px;
line-height: 40px;
}
h2 {
border-bottom: 4px solid #f7f7f7;
padding-top: 10px;
padding-bottom: 2px;
font-size: 145%;
}
h3 {
border-bottom: 2px solid #f7f7f7;
padding-top: 10px;
font-size: 120%;
}
h4 {
border-bottom: 1px solid #f7f7f7;
margin-left: 8px;
font-size: 105%;
}
h5, h6 {
border-bottom: 1px solid #ccc;
font-size: 105%;
}
a {
color: #0033dd;
text-decoration: none;
}
a:hover {
color: #6666ff; }
a:visited {
color: #800080; }
a:visited:hover {
color: #BB00BB; }
a[href^="http:"] {
text-decoration: underline; }
a[href^="https:"] {
text-decoration: underline; }

code > span.kw { color: #555; font-weight: bold; } 
code > span.dt { color: #902000; } 
code > span.dv { color: #40a070; } 
code > span.bn { color: #d14; } 
code > span.fl { color: #d14; } 
code > span.ch { color: #d14; } 
code > span.st { color: #d14; } 
code > span.co { color: #888888; font-style: italic; } 
code > span.ot { color: #007020; } 
code > span.al { color: #ff0000; font-weight: bold; } 
code > span.fu { color: #900; font-weight: bold; } 
code > span.er { color: #a61717; background-color: #e3d2d2; } 
</style>




</head>

<body>




<h1 class="title toc-ignore">RADIOHEAD Example</h1>
<h4 class="author">Shariq Mohammed</h4>
<h4 class="date">2020-12-08</h4>



<p>RADIOHEAD is a package to fit the model proposed in the manuscript: <a href="shariq-mohammed.github.io">S Mohammed</a>, K Bharath, S Kurtek, A Rao, <a href="bayesrx.github.io">V Baladandayuthapani</a>, <em>2020</em>, <em>RADIOHEAD: Radiogenomic Analysis Incorporating Tumor Heterogeneity in Imaging Through Densities</em>.</p>
<p>Example code to run the RADIOHEAD pipeline. This code only shows an example execution of the model using the package RADIOHEAD.</p>
<p>This document contains the following:</p>
<ol style="list-style-type: decimal">
<li><strong>An example to run the RADIOHEAD model using the data available in the package</strong></li>
<li><strong>Pre-processing of the imaging and genomic data</strong>
<ol style="list-style-type: lower-alpha">
<li><strong>Computing pathway scores for a gene set</strong></li>
<li><strong>Pre-processing magnetic resonance imaging scans</strong></li>
</ol></li>
</ol>
<!--- The chunk below will not be executed --->
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1"></a><span class="co"># install the package (devtools package needed)</span></span>
<span id="cb1-2"><a href="#cb1-2"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">require</span>(devtools)) <span class="kw">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3"></a>devtools<span class="op">::</span><span class="kw">install_github</span>(<span class="st">&#39;shariq-mohammed/RADIOHEAD&#39;</span>)</span></code></pre></div>
<p>Load the package</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1"></a><span class="kw">library</span>(RADIOHEAD)</span></code></pre></div>
<div id="example" class="section level1">
<h1>Example</h1>
<p><em>Note that the data is available within the package.</em></p>
<p>Choose the pathway score for the ‘EXOCYTOSIS’ pathway.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1"></a>y =<span class="st"> </span><span class="kw">scale</span>(c_pathway_scores[,<span class="st">&quot;EXOCYTOSIS&quot;</span>], <span class="dt">center =</span> T, <span class="dt">scale =</span> F)</span></code></pre></div>
<p>Choose the principal component scores based on radiomic images.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1"></a>X =<span class="st"> </span><span class="kw">scale</span>(pc_scores, <span class="dt">center =</span> T, <span class="dt">scale =</span> T)</span></code></pre></div>
<p>Choose the group membership labels for the principal components.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1"></a>groups =<span class="st"> </span>pc_groups</span></code></pre></div>
<p>Execute the Gibbs sampling.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1"></a>res =<span class="st"> </span><span class="kw">groupSS</span>(y, X, groups, <span class="dt">Nmcmc=</span><span class="dv">1000</span>)</span>
<span id="cb6-2"><a href="#cb6-2"></a><span class="co">#&gt; [1] &quot;2020-12-08 06:51:37 Iteration 1000&quot;</span></span></code></pre></div>
<p>Identify the significant radiogenomic associaitons between the pathway and the radiomic features after a Bayesian FDR-based variable selection.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1"></a><span class="co"># &#39;associations&#39; outputs the effect size of the significant associations.</span></span>
<span id="cb7-2"><a href="#cb7-2"></a>associations =<span class="st"> </span><span class="kw">fdr_var_selection</span>(res<span class="op">$</span>b, res<span class="op">$</span>x_cnames)</span>
<span id="cb7-3"><a href="#cb7-3"></a>associations</span>
<span id="cb7-4"><a href="#cb7-4"></a><span class="co">#&gt; [1] &quot;No significant associations.&quot;</span></span></code></pre></div>
</div>
<div id="pre-processing-of-the-imaging-and-genomic-data" class="section level1">
<h1>Pre-processing of the imaging and genomic data</h1>
<div id="computing-pathway-scores-for-a-gene-set" class="section level2">
<h2>Computing pathway scores for a gene set</h2>
<p>Below we demonstrate an example to compute pathway scores from a randomly generated gene expression data set. The same process can be employed to compute the pathway scores for any gene expression data.</p>
<p>Install the <code>GSVA</code> package and dependencies. Load <code>GSEABase</code> and <code>GSVA</code> package.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;BiocManager&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) <span class="kw">install.packages</span>(<span class="st">&quot;BiocManager&quot;</span>)</span>
<span id="cb8-2"><a href="#cb8-2"></a><span class="cf">if</span>(<span class="op">!</span><span class="kw">requireNamespace</span>(<span class="st">&quot;GSVA&quot;</span>, <span class="dt">quietly =</span> <span class="ot">TRUE</span>)) BiocManager<span class="op">::</span><span class="kw">install</span>(<span class="st">&quot;GSVA&quot;</span>)</span>
<span id="cb8-3"><a href="#cb8-3"></a><span class="kw">library</span>(GSEABase)</span>
<span id="cb8-4"><a href="#cb8-4"></a><span class="kw">library</span>(GSVA)</span></code></pre></div>
<p>Gene sets can be downloaded from the molecular signature database (<a href="https://www.gsea-msigdb.org/gsea/msigdb/index.jsp">MSigDB</a>). For example, we obtain the gene membership of the EXOCYTOSIS pathway on MSigDB <a href="https://www.gsea-msigdb.org/gsea/msigdb/geneset_page.jsp?geneSetName=GO_EXOCYTOSIS&amp;keywords=exocytosis">from here</a>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1"></a>gene.set =<span class="st"> </span><span class="kw">getGmt</span>(<span class="st">&quot;https://www.gsea-msigdb.org/gsea/msigdb/download_geneset.jsp?geneSetName=GO_EXOCYTOSIS&amp;fileType=gmt&quot;</span>)</span></code></pre></div>
<p>Randomly generate gene expression data for 5 subjects and 1000 genes.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1"></a>gene.expr =<span class="st"> </span><span class="kw">matrix</span>(<span class="kw">rnorm</span>(<span class="dv">5</span><span class="op">*</span><span class="dv">1000</span>), <span class="dt">ncol=</span><span class="dv">5</span>)</span>
<span id="cb10-2"><a href="#cb10-2"></a><span class="kw">colnames</span>(gene.expr) =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;Subject&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>)</span>
<span id="cb10-3"><a href="#cb10-3"></a><span class="kw">rownames</span>(gene.expr) =<span class="st"> </span><span class="kw">paste0</span>(<span class="st">&#39;Gene&#39;</span>, <span class="dv">1</span><span class="op">:</span><span class="dv">1000</span>)</span>
<span id="cb10-4"><a href="#cb10-4"></a></span>
<span id="cb10-5"><a href="#cb10-5"></a><span class="co"># Set row names as the gene names from the pathway</span></span>
<span id="cb10-6"><a href="#cb10-6"></a>gene.ids =<span class="st"> </span>gene.set<span class="op">@</span>.Data[[<span class="dv">1</span>]]<span class="op">@</span>geneIds</span>
<span id="cb10-7"><a href="#cb10-7"></a><span class="kw">rownames</span>(gene.expr)[<span class="dv">1</span><span class="op">:</span><span class="kw">length</span>(gene.ids)] =<span class="st"> </span>gene.ids</span></code></pre></div>
<p>Compute the pathway scores.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1"></a>pathwayScores =<span class="st"> </span><span class="kw">gsva</span>(gene.expr, gene.set, <span class="dt">method=</span><span class="st">&#39;gsva&#39;</span>, <span class="dt">verbose=</span><span class="ot">FALSE</span>)</span>
<span id="cb11-2"><a href="#cb11-2"></a>pathwayScores</span>
<span id="cb11-3"><a href="#cb11-3"></a><span class="co">#&gt;                Subject1    Subject2   Subject3   Subject4    Subject5</span></span>
<span id="cb11-4"><a href="#cb11-4"></a><span class="co">#&gt; GO_EXOCYTOSIS 0.2091517 -0.01411802 0.03157573 -0.1552565 -0.06878825</span></span></code></pre></div>
<p>These pathway scores can be used as response via the <code>groupSS</code> function to fit the model proposed in RADIOHEAD framework.</p>
</div>
<div id="pre-processing-magnetic-resonance-imaging-scans" class="section level2">
<h2>Pre-processing magnetic resonance imaging scans</h2>
<p>The pre-processing of magnetic resonance imaging (MRI) scans involves the following steps</p>
<ul>
<li><em>Step 1</em>: WhiteStripe normalization
<ul>
<li>We perform the WhiteStripe normalization using the <code>whitestripe</code> function directly from the package <a href="https://cran.r-project.org/web/packages/WhiteStripe/index.html"><code>WhiteStripe</code></a>.</li>
</ul></li>
<li><em>Step 2</em>: Obtain tumor voxel values from various groups as defined by different combinations of MRI sequences (T1, T1Gd, T2, FLAIR) and tumor sub-regions (necrosis, edema, and enhancing tumor regions)</li>
<li><em>Step 3</em>: Compute the principal component scores using the Riemannian-geometric framework for each group separately</li>
</ul>
<p>We demonstrate steps 2 and 3 with simple examples (using randomly generated data for five subjects). The same process can be used for real MRI scans.</p>
<p><em>Note:</em> Here we directly work with three dimensional arrays. If the actual MRI scans and the tumor segmentation labels are stored as NIfTI files, then the package <a href="https://cran.r-project.org/web/packages/oro.nifti/index.html"><code>oro.nifti</code></a> can be used to import/load the data.</p>
<p>Generate MRI data of array size 10x10x10 for five subjects by randomly sampling 10x10x10 values from a standard normal distribution. Similarly, generate the corresponding tumor segmentation labels for the five subjects. Here the segmentation labels contain four values, where 0 indicates non-tumor region and 1, 2 and 3 indicate tumor sub-regions.</p>
<p><em>Note:</em> In the following we have not considered multiple MRI sequences. The groups are formed due to the tumor sub-regions. But the same procedure can be repeated for multiple MRI sequences.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1"></a>n =<span class="st"> </span><span class="dv">5</span> <span class="co"># number of subjects</span></span>
<span id="cb12-2"><a href="#cb12-2"></a><span class="cf">for</span>(i <span class="cf">in</span> <span class="dv">1</span><span class="op">:</span><span class="dv">5</span>){</span>
<span id="cb12-3"><a href="#cb12-3"></a>  <span class="kw">assign</span>(<span class="kw">paste0</span>(<span class="st">&#39;mri&#39;</span>,i), <span class="kw">array</span>(<span class="kw">rnorm</span>(<span class="dv">10</span><span class="op">*</span><span class="dv">10</span><span class="op">*</span><span class="dv">10</span>), <span class="dt">dim =</span> <span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">10</span>,<span class="dv">10</span>)))</span>
<span id="cb12-4"><a href="#cb12-4"></a>  <span class="kw">assign</span>(<span class="kw">paste0</span>(<span class="st">&#39;tumor&#39;</span>,i), <span class="kw">array</span>(<span class="kw">rep</span>(<span class="kw">sample</span>(<span class="dv">0</span><span class="op">:</span><span class="dv">3</span>, <span class="dv">4</span>),<span class="kw">c</span>(<span class="dv">125</span>,<span class="dv">125</span>,<span class="dv">125</span>,<span class="dv">1000</span><span class="op">-</span>(<span class="dv">3</span><span class="op">*</span><span class="dv">125</span>))), <span class="dt">dim =</span> <span class="kw">c</span>(<span class="dv">10</span>,<span class="dv">10</span>,<span class="dv">10</span>)))</span>
<span id="cb12-5"><a href="#cb12-5"></a>}</span></code></pre></div>
<div id="step-2" class="section level3">
<h3>Step 2</h3>
<p>Obtain tumor voxel values from three groups as defined by tumor sub-regions (indexed by 1, 2 and 3).</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1"></a><span class="co"># sub-region 1</span></span>
<span id="cb13-2"><a href="#cb13-2"></a>tum<span class="fl">.1</span> =<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span>n,</span>
<span id="cb13-3"><a href="#cb13-3"></a>               <span class="cf">function</span>(i){</span>
<span id="cb13-4"><a href="#cb13-4"></a>                 mri =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;mri&#39;</span>,i)))</span>
<span id="cb13-5"><a href="#cb13-5"></a>                 tumor =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;tumor&#39;</span>,i)))</span>
<span id="cb13-6"><a href="#cb13-6"></a>                 x =<span class="st"> </span><span class="kw">c</span>(mri[tumor<span class="op">==</span><span class="dv">1</span>])</span>
<span id="cb13-7"><a href="#cb13-7"></a>               })</span>
<span id="cb13-8"><a href="#cb13-8"></a></span>
<span id="cb13-9"><a href="#cb13-9"></a><span class="co"># sub-region 2</span></span>
<span id="cb13-10"><a href="#cb13-10"></a>tum<span class="fl">.2</span> =<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span>n,</span>
<span id="cb13-11"><a href="#cb13-11"></a>               <span class="cf">function</span>(i){</span>
<span id="cb13-12"><a href="#cb13-12"></a>                 mri =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;mri&#39;</span>,i)))</span>
<span id="cb13-13"><a href="#cb13-13"></a>                 tumor =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;tumor&#39;</span>,i)))</span>
<span id="cb13-14"><a href="#cb13-14"></a>                 x =<span class="st"> </span><span class="kw">c</span>(mri[tumor<span class="op">==</span><span class="dv">2</span>])</span>
<span id="cb13-15"><a href="#cb13-15"></a>               })</span>
<span id="cb13-16"><a href="#cb13-16"></a></span>
<span id="cb13-17"><a href="#cb13-17"></a><span class="co"># sub-region 3</span></span>
<span id="cb13-18"><a href="#cb13-18"></a>tum<span class="fl">.3</span> =<span class="st"> </span><span class="kw">lapply</span>(<span class="dv">1</span><span class="op">:</span>n,</span>
<span id="cb13-19"><a href="#cb13-19"></a>               <span class="cf">function</span>(i){</span>
<span id="cb13-20"><a href="#cb13-20"></a>                 mri =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;mri&#39;</span>,i)))</span>
<span id="cb13-21"><a href="#cb13-21"></a>                 tumor =<span class="st"> </span><span class="kw">eval</span>(<span class="kw">parse</span>(<span class="dt">text=</span><span class="kw">paste0</span>(<span class="st">&#39;tumor&#39;</span>,i)))</span>
<span id="cb13-22"><a href="#cb13-22"></a>                 x =<span class="st"> </span><span class="kw">c</span>(mri[tumor<span class="op">==</span><span class="dv">3</span>])</span>
<span id="cb13-23"><a href="#cb13-23"></a>               })</span></code></pre></div>
</div>
<div id="step-3" class="section level3">
<h3>Step 3</h3>
<p>Compute the principal component scores for each group separately using the <code>geomPCA</code> function.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1"></a>pc.scores<span class="fl">.1</span> =<span class="st"> </span><span class="kw">geomPCA</span>(tum<span class="fl">.1</span>)</span>
<span id="cb14-2"><a href="#cb14-2"></a>pc.scores<span class="fl">.2</span> =<span class="st"> </span><span class="kw">geomPCA</span>(tum<span class="fl">.2</span>)</span>
<span id="cb14-3"><a href="#cb14-3"></a>pc.scores<span class="fl">.3</span> =<span class="st"> </span><span class="kw">geomPCA</span>(tum<span class="fl">.3</span>)</span></code></pre></div>
<p>These principal component scores can be augmented together (by column) and used to fit the model (via the <code>groupSS</code> function) proposed by in the RADIOHEAD framework. The group indicators (i.e. a label for each column in the augmented matrix) need to be specified appropriately.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1"></a>pc.scores =<span class="st"> </span><span class="kw">cbind</span>(pc.scores<span class="fl">.1</span>, pc.scores<span class="fl">.2</span>, pc.scores<span class="fl">.3</span>)</span>
<span id="cb15-2"><a href="#cb15-2"></a>pc.groups =<span class="st"> </span><span class="kw">rep</span>(<span class="kw">c</span>(<span class="dv">1</span>,<span class="dv">2</span>,<span class="dv">3</span>), <span class="kw">c</span>(<span class="kw">length</span>(pc.scores<span class="fl">.1</span>),</span>
<span id="cb15-3"><a href="#cb15-3"></a>                            <span class="kw">length</span>(pc.scores<span class="fl">.2</span>), <span class="kw">length</span>(pc.scores<span class="fl">.3</span>)))</span></code></pre></div>
</div>
</div>
</div>



<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
